src/main/java/org/kohsuke/github/GHBranch.java

```diff
@@ -192,6 +192,19 @@ public GHCommit merge(String head, String commitMessage) throws IOException {
         return result;
     }
 
+    /**
+     * Rename this branch.
+     *
+     * @param name
+     *            the name
+     * @throws IOException
+     *             the io exception
+     */
+    @CheckForNull
+    public void rename(String name) throws IOException {
```

Comment:
The endpoint returns a `GHBranch`, so should the method. 
```suggestion
     *             the io exception
     * @see https://docs.github.com/en/rest/branches/branches?apiVersion=2022-11-28#rename-a-branch
     */
    public GHBranch rename(String name) throws IOException {
```